enhancements/hypershift/kubevirt-static-ip-address-management.md

```diff
@@ -0,0 +1,690 @@
+
+### Drawbacks
+
+- Adds complexity to the hypershift kubevirt controllers:
+  - implementing an ip pool mechanism
+  - Needs to parse and understand the openstack (configdrive) or netplan (nocloud) network config format
+- Fix depend on a coreos change:
+  - It will take some time for the RPM to reach targeted RHCOS, also we will have to plan backports there.
+
+## Alternatives (Not Implemented)
+
+### Putting the network configuration inside the AdditionalNetworks structs
+
+Add `networkConfig` to the AdditionalNetwork struct
+
+```yaml
+type AdditionalNetwork struct{
+   Name string
+   NetworkConfig ... <- this is new 
+}
+```
+
+Pros:
+- We re-use the current addtional network concept without introducing bootstrap one
+- Allow to use the ip allocator at secondary networks too
+
+Cons:
+- Introduce more complexity, since it opens the door to do network configuration for all the networks
+- We don't really know if users may need an hypershift ipam mechanism for secondaries.
+- Users can configure common network configuration with ignition
+
+### Store all the ignition config directly at a secret instead of retrieve it using the ignition server
+
+Currently hypershift workers retrieve the ignition config from an ignition server, so they need to have network configured before that, an alternative
+would be to directly put the whole ignition config directly at the secret referenced by the VM so accessing ignition server is not needed.
+
+**Why not chosen**: The kubernetes resources has a size limit that has already been reached by hypershift when storing the whole ignition config there [Jira](https://issues.redhat.com//browse/OCPBUGS-60148), also it means changing how a hypershift fundamental part of the bootstrap process works.
+
+### Use the ovn-kubernetes localnet IPAM mechanism
+
+With this solution IPAM configuration will be done at the ovn-kubernetes localnet user defined network instead of the hypershift kubevirt nodepoo
+
+Missing features from current localnet implementation:
+- There is no official support for localnet + IPAM
+- localnet is missing gateway and dns configuration
+- There is no out of the box solution for IPv6 RAs (there is no OVN logical router here)
+
+Pro:
+- A lot of the missing logic is already present for primary UDN layer2 as part of the IP migration from MTV, only missing part is name resolution server.
+- This works for VMs and pods, so it would be possible in the future to also put the control plane under same network and IPAM.
+- It will be supported for ovn-kubernetes team, not only hypershift kubevirt maintainers.
+- There is a center place to allocate IPs so there will be no overlap between hosted clusters
+
+Cons:
+- No clear solution for implementing IPv6 RAs since there is no logical router.
+- Contributing this feature to ovn-kubernetes may take months if not years and a lot of effort, 
+  since it's not really aligned with ovn-kubernetes maintainers priorities.
```

Comment:
This is a prioritization issue and as such I don't think it belongs to a technical enhancement